GitHub App trigger options new page (#9198)
circleci/circleci-docs: jekyll/_cci2/orchestration-cookbook.adoc

---
contentTags:
  platform:
  - Cloud
---
= Orchestration cookbook
:page-layout: classic-docs
:page-liquid:
:page-description: A cookbook of common pipeline, workflow and job execution goals and patterns
:icons: font
:experimental:

This cookbook contains "recipes" for common pipeline, workflow and job execution goals and patterns.

== Trigger a pipeline when a PR is marked ready for review

=== Goal

> You may wish to only build your project, run tests, and other work when a pull request is ready to be taken out of its draft state -- when the PR is marked as ready for review. To do this, set up a trigger for your pipeline that triggers when a pull request is marked ready for review, as follows:

Reviewer: ```suggestion
You may wish to run a specific set of tests only when a pull request is ready to be taken out of its draft state -- when the PR is marked as ready for review. To do this, set up a trigger for your pipeline that triggers when a pull request is marked ready for review, as follows:
```

=== Steps

. Set up a GitHub App pipeline you want to trigger when a PR is marked ready for review. Steps for setting up a pipeline are available in the xref:pipelines#add-or-edit-a-pipeline[Pipelines and triggers overview] page.
+
.Add pipeline options for GitHub App
image::add-pipeline.png[Add pipeline options for GitHub App]
. Set up a trigger for your pipeline and select *PR marked ready for review* under the *Run on* menu. Steps for setting up a trigger are available in the xref:pipelines#add-a-trigger[Pipelines and triggers overview] page.
+
.Run on trigger options for GitHub App triggers

> image::triggers/run-on-open.png[Run on trigger options for GitHub App triggers]

Reviewer: Reminder to myself to check this section in preview, I'm finding it difficult to undersand the different subsections

Reviewer: Checked - this looks good!

=== Notes

> * This option is only available for GitHub App pipelines. If you have a GitHub OAuth integration you can install the GitHub App into your GitHub org and then create GitHub App pipelines.

Reviewer: Can we call out "this is a quick one-time action" + link them to the docs?

== Trigger a pipeline when a PR is merged

=== Goal

You may wish to trigger a specific pipeline when a pull request is merged, for example, to run clean-up/teardown scripts. To do this, set up a trigger for a "clean-up" pipeline that is triggered when a pull request is merged.

=== Steps
. Set up a GitHub App pipeline you want to trigger when a PR is merged. Steps for setting up a pipeline are available in the xref:pipelines#add-or-edit-a-pipeline[Pipelines and triggers overview] page.
+
.Add pipeline options for GitHub App
image::add-pipeline.png[Add pipeline options for GitHub App]
. Set up a trigger for your pipeline and select *PR merged* under the *Run on* menu. Steps for setting up a trigger are available in the xref:pipelines#add-a-trigger[Pipelines and triggers overview] page.
+
.Run on trigger options for GitHub App triggers
image::triggers/run-on-open.png[Run on trigger options for GitHub App triggers]

=== Notes

> * This option is only available for GitHub App pipelines. If you have a GitHub OAuth integration you can install the GitHub App into your GitHub org and then create GitHub App pipelines.

Reviewer: Same as above
